File: nomad-samples/admin/js/basic-live-channel/live-operator/complete-segment.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";

export default async function completeSegment(AUTH_TOKEN, ID, RELATED_CONTENT_IDS, TAG_IDS) {
    // Create header for the request
    const HEADERS = new Headers();
    HEADERS.append("Authorization", `Bearer ${AUTH_TOKEN}`);
    HEADERS.append("Content-Type", "application/json");

    // Create body for the request
    const BODY = {
        liveOperatorId: ID
    };

    if (RELATED_CONTENT_IDS[0] !== "") {
        BODY.relatedContent = RELATED_CONTENT_IDS.map(id => ({ id }));
    }
    if (TAG_IDS[0] !== "") {
        BODY.tags = TAG_IDS.map(id => ({ id }));
    }

    // Send the request
    const RESPONSE = await fetch(`${prjConstants.ADMIN_API_URL}/admin/liveOperator/${ID}/completeSegment`, {
        method: "POST",
        headers: HEADERS,
        body: JSON.stringify(BODY)
    });

    // Check for success
    if (RESPONSE && RESPONSE.ok) {
        return await RESPONSE.json();
    }

    await apiExceptionHandler(RESPONSE, `Completing Segment ${ID} failed`);
}

File: nomad-samples/admin/py/basic-live-channel-1.1/live_operator/complete_segment.py
from constants.project_constants import *
from exceptions.api_exception_handler import *

import requests, json

def complete_segment(AUTH_TOKEN, ID, RELATED_CONTENT_IDS, TAG_IDS):

    HEADERS = {
        "Authorization": "Bearer " +  AUTH_TOKEN,
        "Content-Type": "application/json"
    }

    BODY = {
        "liveOperatorId": ID,
    }
    if RELATED_CONTENT_IDS[0] != "":
        BODY["relatedContent"] = [] 
        for ID in RELATED_CONTENT_IDS:
            BODY["relatedContent"].append({ "id": ID }),
    if TAG_IDS[0] != "":
        BODY["tags"] = []
        for ID in TAG_IDS:
            BODY["tags"].append({"id": ID })

    try:
        RESPONSE = requests.post(f"{ADMIN_URL}/admin/liveOperator/{ID}/completeSegment", headers= HEADERS, data= json.dumps(BODY))

        # If not found return None
        if (not RESPONSE.ok):
            raise Exception()
    
    except:
        api_exception_handler(RESPONSE, f"Completing segment for Live Channel {ID} failed")